systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java

```diff
@@ -348,4 +349,57 @@ void testCruiseControlIntraBrokerBalancingWithoutSpecifyingJBODStorage(Extension
+
+        LOGGER.info("Scaling Kafka up to {}", scaleTo);
+
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), kafka -> kafka.getSpec().getKafka().setReplicas(scaleTo), testStorage.getNamespaceName());
+        kafkaPods = RollingUpdateUtils.waitForComponentScaleUpOrDown(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), scaleTo, kafkaPods);
+
+        LOGGER.info("Creating KafkaRebalance with add_brokers mode");
+
+        // when using add_brokers mode, we can hit `ProposalReady` right after KR creation - that's why `waitReady` is set to `false` here
+        resourceManager.createResource(extensionContext, false,
+            KafkaRebalanceTemplates.kafkaRebalance(testStorage.getClusterName())
+                .editOrNewSpec()
+                    .withMode(KafkaRebalanceMode.ADD_BROKERS)
+                    .withBrokers(3)
+                .endSpec()
+                .build()
+        );
+
+        KafkaRebalanceUtils.waitForKafkaRebalanceCustomResourceState(testStorage.getNamespaceName(),  testStorage.getClusterName(), KafkaRebalanceState.ProposalReady);
+        KafkaRebalanceUtils.doRebalancingProcess(new Reconciliation("test", KafkaRebalance.RESOURCE_KIND, testStorage.getNamespaceName(), testStorage.getClusterName()), testStorage.getNamespaceName(), testStorage.getClusterName());
+
+        KafkaRebalanceResource.kafkaRebalanceClient().inNamespace(testStorage.getNamespaceName()).withName(testStorage.getClusterName()).delete();
+
+        LOGGER.info("Creating KafkaRebalance with remove_brokers mode - it needs to be done before actual scaling down of Kafka pods");
+
+        // when using remove_brokers mode, we can hit `ProposalReady` right after KR creation - that's why `waitReady` is set to `false` here
+        resourceManager.createResource(extensionContext, false,
+            KafkaRebalanceTemplates.kafkaRebalance(testStorage.getClusterName())
+                .editOrNewSpec()
+                    .withMode(KafkaRebalanceMode.REMOVE_BROKERS)
+                    .withBrokers(3)
+                .endSpec()
+                .build()
+        );
+
+        KafkaRebalanceUtils.waitForKafkaRebalanceCustomResourceState(testStorage.getNamespaceName(),  testStorage.getClusterName(), KafkaRebalanceState.ProposalReady);
+        KafkaRebalanceUtils.doRebalancingProcess(new Reconciliation("test", KafkaRebalance.RESOURCE_KIND, testStorage.getNamespaceName(), testStorage.getClusterName()), testStorage.getNamespaceName(), testStorage.getClusterName());
+
+        LOGGER.info("Scaling Kafka down to {}", initialReplicas);
+
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), kafka -> kafka.getSpec().getKafka().setReplicas(initialReplicas), testStorage.getNamespaceName());
+        RollingUpdateUtils.waitForComponentScaleUpOrDown(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), initialReplicas, kafkaPods);
```

Comment:
Shouldn't we have here some check that everything is fine?

Reply:
What do you prefer?
The `remove_broker` rebalancing is done before scaling down and the `waitForComponentScaleUpOrDown` ensures that pods are ready.
Maybe we can do full rebalance, but it will add more execution time and I think that it is not necessary here.

Reply:
The check if the KafkaRebalance is in `Ready` state is inside the `doRebalancingProcess` method - any other check is not needed (as discussed offline)